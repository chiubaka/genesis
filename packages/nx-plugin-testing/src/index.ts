export * from "./setup";
export * from "./testingWorkspace";
export * from "./verdaccio";
