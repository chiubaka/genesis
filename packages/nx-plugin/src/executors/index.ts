export * from "./nxPluginE2e";
