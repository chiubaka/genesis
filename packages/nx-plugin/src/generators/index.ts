export * from "./linting";
export * from "./preset";
