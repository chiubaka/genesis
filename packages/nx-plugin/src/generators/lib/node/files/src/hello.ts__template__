import crypto from "node:crypto";
import fs from "node:fs";
import os from "node:os";
import path from "node:path";

export const hello = "Hello, world!";

export function writeHelloWorld() {
  const randomSuffix = crypto.randomBytes(48).toString("base64url");
  const fileName = `hello-${randomSuffix}.txt`;

  const tmpDir = os.tmpdir();
  const filePath = path.join(tmpDir, fileName);

  // eslint-disable-next-line security/detect-non-literal-fs-filename
  fs.writeFileSync(filePath, hello);

  return filePath;
}

export class Greeter {
  public hello() {
    return hello;
  }
}
