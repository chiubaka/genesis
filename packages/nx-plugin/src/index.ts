export * from "./generators";
export * from "./utils";
