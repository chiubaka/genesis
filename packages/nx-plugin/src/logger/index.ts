import { Logger, LoggerOptions, LogLevel } from "./Logger";
import { WinstonLogger } from "./winston";

export const createLogger = (
  name: string,
  options: Partial<LoggerOptions> = {},
): Logger => {
  return new WinstonLogger({
    level: LogLevel.Info,
    ...options,
    output: {
      console: true,
      file: [
        {
          path: `${name}.log`,
        },
      ],
      ...options.output,
    },
  });
};

export const generatorLogger = createLogger("generator");
