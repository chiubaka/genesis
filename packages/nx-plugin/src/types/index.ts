export * from "./codecov";
export * from "./dockerCompose";
export * from "./npm";
export * from "./nx";
export * from "./verdaccio";
export * from "./yarn";
