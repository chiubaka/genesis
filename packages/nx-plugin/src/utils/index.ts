export * from "./addDependenciesToPackageJson";
export * from "./exec";
export * from "./getProjectDir";
export * from "./git";
export * from "./github";
export * from "./Project";
export * from "./projectNameCasings";
export * from "./tasks";
