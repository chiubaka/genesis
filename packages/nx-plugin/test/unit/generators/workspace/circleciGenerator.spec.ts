import { Tree } from "@nx/devkit";
import { createTreeWithEmptyWorkspace } from "@nx/devkit/testing";

import { CircleCiConfig, circleciGenerator, readYaml } from "../../../../src";

describe("circleciGenerator", () => {
  let tree: Tree;

  beforeAll(() => {
    tree = createTreeWithEmptyWorkspace();
    circleciGenerator(tree);
  });

  it("creates a .circleci/config.yml file", () => {
    expect(tree.exists(".circleci/config.yml")).toBe(true);
  });

  describe(".circleci/config.yml", () => {
    let config: CircleCiConfig;

    beforeAll(() => {
      config = readYaml<CircleCiConfig>(tree, ".circleci/config.yml");
    });

    it("matches snapshot", () => {
      expect(config).toMatchSnapshot();
    });

    it("includes the chiubaka/circleci-orb", () => {
      expect(config.orbs?.chiubaka).toContain("chiubaka/circleci-orb");
    });

    it("is a setup config", () => {
      expect(config.setup).toBe(true);
    });

    it("defines a dynamic-setup workflow", () => {
      expect(config.workflows["dynamic-setup"]).toBeDefined();
    });
  });

  describe(".circleci/react-native.template.yml", () => {
    let config: CircleCiConfig;

    beforeAll(() => {
      config = readYaml<CircleCiConfig>(
        tree,
        ".circleci/react-native.template.yml",
      );
    });

    it("matches snapshot", () => {
      expect(config).toMatchSnapshot();
    });

    it("defines a lint-build-test-e2e-deploy workflow", () => {
      expect(config.workflows["lint-build-test-e2e-deploy"]).toBeDefined();
    });
  });

  describe(".circleci/js.template.yml", () => {
    let config: CircleCiConfig;

    beforeAll(() => {
      config = readYaml<CircleCiConfig>(tree, ".circleci/js.template.yml");
    });

    it("matches snapshot", () => {
      expect(config).toMatchSnapshot();
    });

    it("defines a lint-build-test-e2e-deploy workflow", () => {
      expect(config.workflows["lint-build-test-e2e-deploy"]).toBeDefined();
    });
  });
});
