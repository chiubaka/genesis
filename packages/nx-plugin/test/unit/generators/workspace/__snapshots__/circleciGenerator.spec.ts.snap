// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`circleciGenerator .circleci/config.yml matches snapshot 1`] = `
{
  "constants": {
    "semver-regex": "/^(\\w+-)+v(0|[1-9]\\d*)\\.(0|[1-9]\\d*)\\.(0|[1-9]\\d*)(?:-((?:0|[1-9]\\d*|\\d*[a-zA-Z-][0-9a-zA-Z-]*)(?:\\.(?:0|[1-9]\\d*|\\d*[a-zA-Z-][0-9a-zA-Z-]*))*))?(?:\\+([0-9a-zA-Z-]+(?:\\.[0-9a-zA-Z-]+)*))?$/",
  },
  "filters": {
    "semver-tags": {
      "tags": {
        "only": "/^(\\w+-)+v(0|[1-9]\\d*)\\.(0|[1-9]\\d*)\\.(0|[1-9]\\d*)(?:-((?:0|[1-9]\\d*|\\d*[a-zA-Z-][0-9a-zA-Z-]*)(?:\\.(?:0|[1-9]\\d*|\\d*[a-zA-Z-][0-9a-zA-Z-]*))*))?(?:\\+([0-9a-zA-Z-]+(?:\\.[0-9a-zA-Z-]+)*))?$/",
      },
    },
  },
  "orbs": {
    "chiubaka": "chiubaka/circleci-orb@dev:alpha",
  },
  "setup": true,
  "version": 2.1,
  "workflows": {
    "dynamic-setup": {
      "jobs": [
        {
          "chiubaka/dynamic-setup": {
            "filters": {
              "tags": {
                "only": "/^(\\w+-)+v(0|[1-9]\\d*)\\.(0|[1-9]\\d*)\\.(0|[1-9]\\d*)(?:-((?:0|[1-9]\\d*|\\d*[a-zA-Z-][0-9a-zA-Z-]*)(?:\\.(?:0|[1-9]\\d*|\\d*[a-zA-Z-][0-9a-zA-Z-]*))*))?(?:\\+([0-9a-zA-Z-]+(?:\\.[0-9a-zA-Z-]+)*))?$/",
              },
            },
          },
        },
      ],
    },
  },
}
`;

exports[`circleciGenerator .circleci/js.template.yml matches snapshot 1`] = `
{
  "constants": {
    "semver-regex": "/^(\\w+-)+v(0|[1-9]\\d*)\\.(0|[1-9]\\d*)\\.(0|[1-9]\\d*)(?:-((?:0|[1-9]\\d*|\\d*[a-zA-Z-][0-9a-zA-Z-]*)(?:\\.(?:0|[1-9]\\d*|\\d*[a-zA-Z-][0-9a-zA-Z-]*))*))?(?:\\+([0-9a-zA-Z-]+(?:\\.[0-9a-zA-Z-]+)*))?$/",
  },
  "filters": {
    "ignore-branches": {
      "branches": {
        "ignore": "/.*/",
      },
    },
    "semver-tags": {
      "tags": {
        "only": "/^(\\w+-)+v(0|[1-9]\\d*)\\.(0|[1-9]\\d*)\\.(0|[1-9]\\d*)(?:-((?:0|[1-9]\\d*|\\d*[a-zA-Z-][0-9a-zA-Z-]*)(?:\\.(?:0|[1-9]\\d*|\\d*[a-zA-Z-][0-9a-zA-Z-]*))*))?(?:\\+([0-9a-zA-Z-]+(?:\\.[0-9a-zA-Z-]+)*))?$/",
      },
    },
  },
  "orbs": {
    "chiubaka": "chiubaka/circleci-orb@0.11.4",
  },
  "version": 2.1,
  "workflows": {
    "lint-build-test-e2e-deploy": {
      "jobs": [
        {
          "chiubaka/lint": {
            "filters": {
              "tags": {
                "only": "/^(\\w+-)+v(0|[1-9]\\d*)\\.(0|[1-9]\\d*)\\.(0|[1-9]\\d*)(?:-((?:0|[1-9]\\d*|\\d*[a-zA-Z-][0-9a-zA-Z-]*)(?:\\.(?:0|[1-9]\\d*|\\d*[a-zA-Z-][0-9a-zA-Z-]*))*))?(?:\\+([0-9a-zA-Z-]+(?:\\.[0-9a-zA-Z-]+)*))?$/",
              },
            },
            "name": "lint",
          },
        },
        {
          "chiubaka/build": {
            "filters": {
              "tags": {
                "only": "/^(\\w+-)+v(0|[1-9]\\d*)\\.(0|[1-9]\\d*)\\.(0|[1-9]\\d*)(?:-((?:0|[1-9]\\d*|\\d*[a-zA-Z-][0-9a-zA-Z-]*)(?:\\.(?:0|[1-9]\\d*|\\d*[a-zA-Z-][0-9a-zA-Z-]*))*))?(?:\\+([0-9a-zA-Z-]+(?:\\.[0-9a-zA-Z-]+)*))?$/",
              },
            },
            "name": "build",
          },
        },
        {
          "chiubaka/test": {
            "filters": {
              "tags": {
                "only": "/^(\\w+-)+v(0|[1-9]\\d*)\\.(0|[1-9]\\d*)\\.(0|[1-9]\\d*)(?:-((?:0|[1-9]\\d*|\\d*[a-zA-Z-][0-9a-zA-Z-]*)(?:\\.(?:0|[1-9]\\d*|\\d*[a-zA-Z-][0-9a-zA-Z-]*))*))?(?:\\+([0-9a-zA-Z-]+(?:\\.[0-9a-zA-Z-]+)*))?$/",
              },
            },
            "name": "test",
          },
        },
        {
          "chiubaka/e2e": {
            "filters": {
              "tags": {
                "only": "/^(\\w+-)+v(0|[1-9]\\d*)\\.(0|[1-9]\\d*)\\.(0|[1-9]\\d*)(?:-((?:0|[1-9]\\d*|\\d*[a-zA-Z-][0-9a-zA-Z-]*)(?:\\.(?:0|[1-9]\\d*|\\d*[a-zA-Z-][0-9a-zA-Z-]*))*))?(?:\\+([0-9a-zA-Z-]+(?:\\.[0-9a-zA-Z-]+)*))?$/",
              },
            },
            "name": "e2e",
          },
        },
        {
          "chiubaka/deploy": {
            "context": [
              "npm-publishing",
            ],
            "filters": {
              "branches": {
                "ignore": "/.*/",
              },
              "tags": {
                "only": "/^(\\w+-)+v(0|[1-9]\\d*)\\.(0|[1-9]\\d*)\\.(0|[1-9]\\d*)(?:-((?:0|[1-9]\\d*|\\d*[a-zA-Z-][0-9a-zA-Z-]*)(?:\\.(?:0|[1-9]\\d*|\\d*[a-zA-Z-][0-9a-zA-Z-]*))*))?(?:\\+([0-9a-zA-Z-]+(?:\\.[0-9a-zA-Z-]+)*))?$/",
              },
            },
            "name": "deploy",
            "requires": [
              "lint",
              "build",
              "test",
              "e2e",
            ],
          },
        },
      ],
    },
  },
}
`;

exports[`circleciGenerator .circleci/react-native.template.yml matches snapshot 1`] = `
{
  "constants": {
    "android-job-params": {
      "build-tools-version": "<< pipeline.parameters.android-emulator-build-tools-version >>",
      "platform-version": "<< pipeline.parameters.android-emulator-platform-version >>",
      "setup-apps-steps": "$SETUP_ANDROID_APPS_STEPS",
    },
    "android-semver-regex": "$ANDROID_SEMVER_REGEX",
    "ios-job-params": {
      "context": [
        "ios-deployment",
      ],
      "setup-apps-steps": "$SETUP_IOS_APPS_STEPS",
      "simulator-device": "<< pipeline.parameters.ios-simulator-device >>",
      "simulator-version": "<< pipeline.parameters.ios-simulator-version >>",
      "xcode-version": "<< pipeline.parameters.xcode-version >>",
    },
    "ios-semver-regex": "$IOS_SEMVER_REGEX",
    "semver-regex": "/^(\\w+-)+v(0|[1-9]\\d*)\\.(0|[1-9]\\d*)\\.(0|[1-9]\\d*)(?:-((?:0|[1-9]\\d*|\\d*[a-zA-Z-][0-9a-zA-Z-]*)(?:\\.(?:0|[1-9]\\d*|\\d*[a-zA-Z-][0-9a-zA-Z-]*))*))?(?:\\+([0-9a-zA-Z-]+(?:\\.[0-9a-zA-Z-]+)*))?$/",
  },
  "filters": {
    "android-semver-tags": {
      "tags": {
        "only": "$ANDROID_SEMVER_REGEX",
      },
    },
    "ignore-android-semver-tags": {
      "tags": {
        "ignore": "$ANDROID_SEMVER_REGEX",
      },
    },
    "ignore-branches": {
      "branches": {
        "ignore": "/.*/",
      },
    },
    "ignore-ios-semver-tags": {
      "tags": {
        "ignore": "$IOS_SEMVER_REGEX",
      },
    },
    "ios-semver-tags": {
      "tags": {
        "only": "$IOS_SEMVER_REGEX",
      },
    },
    "semver-tags": {
      "tags": {
        "only": "/^(\\w+-)+v(0|[1-9]\\d*)\\.(0|[1-9]\\d*)\\.(0|[1-9]\\d*)(?:-((?:0|[1-9]\\d*|\\d*[a-zA-Z-][0-9a-zA-Z-]*)(?:\\.(?:0|[1-9]\\d*|\\d*[a-zA-Z-][0-9a-zA-Z-]*))*))?(?:\\+([0-9a-zA-Z-]+(?:\\.[0-9a-zA-Z-]+)*))?$/",
      },
    },
  },
  "orbs": {
    "chiubaka": "chiubaka/circleci-orb@dev:alpha",
  },
  "parameters": {
    "android-emulator-build-tools-version": {
      "default": "34.0.0",
      "description": "The build tools version to use for the Android emulator when required",
      "type": "string",
    },
    "android-emulator-platform-version": {
      "default": "android-34",
      "description": "The platform version to use for the Android emulator when required",
    },
    "build-android": {
      "default": false,
      "description": "Whether or not to enable the build-android job",
      "type": "boolean",
    },
    "build-ios": {
      "default": false,
      "description": "Whether or not to enable the build-ios job",
      "type": "boolean",
    },
    "e2e-android": {
      "default": false,
      "description": "Whether or not to enable the e2e-android job",
      "type": "boolean",
    },
    "e2e-ios": {
      "default": false,
      "description": "Whether or not to enable the e2e-ios job",
      "type": "boolean",
    },
    "ios-simulator-device": {
      "default": "iPhone 14",
      "description": "The iOS device to use for the simulator when required",
      "type": "string",
    },
    "ios-simulator-version": {
      "default": "16.4",
      "description": "The OS version to use for the simulator when required",
      "type": "string",
    },
    "test-android": {
      "default": false,
      "description": "Whether or not to enable the test-android job",
      "type": "boolean",
    },
    "test-ios": {
      "default": false,
      "description": "Whether or not to enable the test-ios job",
      "type": "boolean",
    },
    "xcode-version": {
      "default": "14.3.1",
      "description": "The version of Xcode to use with for macOS executors",
      "type": "string",
    },
  },
  "version": 2.1,
  "workflows": {
    "lint-build-test-e2e-deploy": {
      "jobs": [
        {
          "chiubaka/lint": {
            "filters": {
              "tags": {
                "only": "/^(\\w+-)+v(0|[1-9]\\d*)\\.(0|[1-9]\\d*)\\.(0|[1-9]\\d*)(?:-((?:0|[1-9]\\d*|\\d*[a-zA-Z-][0-9a-zA-Z-]*)(?:\\.(?:0|[1-9]\\d*|\\d*[a-zA-Z-][0-9a-zA-Z-]*))*))?(?:\\+([0-9a-zA-Z-]+(?:\\.[0-9a-zA-Z-]+)*))?$/",
              },
            },
            "name": "lint",
          },
        },
        {
          "chiubaka/build": {
            "filters": {
              "tags": {
                "only": "/^(\\w+-)+v(0|[1-9]\\d*)\\.(0|[1-9]\\d*)\\.(0|[1-9]\\d*)(?:-((?:0|[1-9]\\d*|\\d*[a-zA-Z-][0-9a-zA-Z-]*)(?:\\.(?:0|[1-9]\\d*|\\d*[a-zA-Z-][0-9a-zA-Z-]*))*))?(?:\\+([0-9a-zA-Z-]+(?:\\.[0-9a-zA-Z-]+)*))?$/",
              },
            },
            "name": "build",
          },
        },
        {
          "chiubaka/test": {
            "filters": {
              "tags": {
                "only": "/^(\\w+-)+v(0|[1-9]\\d*)\\.(0|[1-9]\\d*)\\.(0|[1-9]\\d*)(?:-((?:0|[1-9]\\d*|\\d*[a-zA-Z-][0-9a-zA-Z-]*)(?:\\.(?:0|[1-9]\\d*|\\d*[a-zA-Z-][0-9a-zA-Z-]*))*))?(?:\\+([0-9a-zA-Z-]+(?:\\.[0-9a-zA-Z-]+)*))?$/",
              },
            },
            "name": "test",
          },
        },
        {
          "chiubaka/e2e": {
            "filters": {
              "tags": {
                "only": "/^(\\w+-)+v(0|[1-9]\\d*)\\.(0|[1-9]\\d*)\\.(0|[1-9]\\d*)(?:-((?:0|[1-9]\\d*|\\d*[a-zA-Z-][0-9a-zA-Z-]*)(?:\\.(?:0|[1-9]\\d*|\\d*[a-zA-Z-][0-9a-zA-Z-]*))*))?(?:\\+([0-9a-zA-Z-]+(?:\\.[0-9a-zA-Z-]+)*))?$/",
              },
            },
            "name": "e2e",
            "requires": [
              "build",
            ],
          },
        },
        {
          "chiubaka/deploy": {
            "filters": {
              "branches": {
                "ignore": "/.*/",
              },
              "tags": {
                "ignore": "$ANDROID_SEMVER_REGEX",
              },
            },
            "name": "deploy",
            "requires": [
              "build",
              "test",
              "e2e",
            ],
          },
        },
        {
          "chiubaka/build-android": {
            "build-tools-version": "<< pipeline.parameters.android-emulator-build-tools-version >>",
            "enabled": "<< pipeline.parameters.build-android >>",
            "filters": {
              "tags": {
                "only": "$ANDROID_SEMVER_REGEX",
              },
            },
            "name": "build-android",
            "platform-version": "<< pipeline.parameters.android-emulator-platform-version >>",
            "requires": [
              "build",
            ],
            "setup-apps-steps": "$SETUP_ANDROID_APPS_STEPS",
          },
        },
        {
          "chiubaka/build-ios": {
            "context": [
              "ios-deployment",
            ],
            "enabled": "<< pipeline.parameters.build-ios >>",
            "filters": {
              "tags": {
                "only": "$ANDROID_SEMVER_REGEX",
              },
            },
            "name": "build-ios",
            "requires": [
              "build",
            ],
            "setup-apps-steps": "$SETUP_IOS_APPS_STEPS",
            "simulator-device": "<< pipeline.parameters.ios-simulator-device >>",
            "simulator-version": "<< pipeline.parameters.ios-simulator-version >>",
            "xcode-version": "<< pipeline.parameters.xcode-version >>",
          },
        },
        {
          "chiubaka/test-android": {
            "build-tools-version": "<< pipeline.parameters.android-emulator-build-tools-version >>",
            "enabled": "<< pipeline.parameters.test-android >>",
            "filters": {
              "tags": {
                "only": "$ANDROID_SEMVER_REGEX",
              },
            },
            "name": "test-android",
            "platform-version": "<< pipeline.parameters.android-emulator-platform-version >>",
            "requires": [
              "build",
              "build-android",
              "test",
              "e2e",
            ],
            "setup-apps-steps": "$SETUP_ANDROID_APPS_STEPS",
          },
        },
        {
          "chiubaka/test-ios": {
            "context": [
              "ios-deployment",
            ],
            "enabled": "<< pipeline.parameters.test-ios >>",
            "filters": {
              "tags": {
                "only": "$ANDROID_SEMVER_REGEX",
              },
            },
            "name": "test-ios",
            "requires": [
              "build",
              "build-ios",
              "test",
              "e2e",
            ],
            "setup-apps-steps": "$SETUP_IOS_APPS_STEPS",
            "simulator-device": "<< pipeline.parameters.ios-simulator-device >>",
            "simulator-version": "<< pipeline.parameters.ios-simulator-version >>",
            "xcode-version": "<< pipeline.parameters.xcode-version >>",
          },
        },
        {
          "chiubaka/e2e-android": {
            "build-tools-version": "<< pipeline.parameters.android-emulator-build-tools-version >>",
            "enabled": "<< pipeline.parameters.e2e-android >>",
            "filters": {
              "tags": {
                "only": "$ANDROID_SEMVER_REGEX",
              },
            },
            "name": "e2e-android",
            "platform-version": "<< pipeline.parameters.android-emulator-platform-version >>",
            "requires": [
              "build-android",
              "test-android",
            ],
            "setup-apps-steps": "$SETUP_ANDROID_APPS_STEPS",
          },
        },
        {
          "chiubaka/e2e-ios": {
            "context": [
              "ios-deployment",
            ],
            "enabled": "<< pipeline.parameters.e2e-ios >>",
            "filters": {
              "tags": {
                "only": "$ANDROID_SEMVER_REGEX",
              },
            },
            "name": "e2e-ios",
            "requires": [
              "build-ios",
              "test-ios",
            ],
            "setup-apps-steps": "$SETUP_IOS_APPS_STEPS",
            "simulator-device": "<< pipeline.parameters.ios-simulator-device >>",
            "simulator-version": "<< pipeline.parameters.ios-simulator-version >>",
            "xcode-version": "<< pipeline.parameters.xcode-version >>",
          },
        },
        {
          "chiubaka/deploy-android": {
            "build-tools-version": "<< pipeline.parameters.android-emulator-build-tools-version >>",
            "filters": {
              "branches": {
                "ignore": "/.*/",
              },
              "tags": {
                "only": "$ANDROID_SEMVER_REGEX",
              },
            },
            "name": "deploy-android",
            "platform-version": "<< pipeline.parameters.android-emulator-platform-version >>",
            "requires": [
              "e2e-android",
            ],
            "setup-apps-steps": "$SETUP_ANDROID_APPS_STEPS",
          },
        },
        {
          "chiubaka/deploy-ios": {
            "context": [
              "ios-deployment",
            ],
            "filters": {
              "branches": {
                "ignore": "/.*/",
              },
              "tags": {
                "only": "$IOS_SEMVER_REGEX",
              },
            },
            "name": "deploy-ios",
            "requires": [
              "e2e-ios",
            ],
            "setup-apps-steps": "$SETUP_IOS_APPS_STEPS",
            "simulator-device": "<< pipeline.parameters.ios-simulator-device >>",
            "simulator-version": "<< pipeline.parameters.ios-simulator-version >>",
            "xcode-version": "<< pipeline.parameters.xcode-version >>",
          },
        },
      ],
    },
  },
}
`;
